sky/execution.py

```diff
@@ -702,11 +702,18 @@ def _maybe_translate_local_file_mounts_and_sync_up(
     if copy_mounts is None:
         copy_mounts = {}
 
-    has_local_source_paths = (task.workdir is not None) or copy_mounts
-    if has_local_source_paths:
-        logger.info(
-            f'{colorama.Fore.YELLOW}Translating file_mounts with local '
-            f'source paths to SkyPilot Storage...{colorama.Style.RESET_ALL}')
+    has_local_source_paths_workdir = task.workdir is not None
+    has_local_source_paths_file_mounts = copy_mounts is not None
```

Comment:
L703 sets `copy_mounts = {}` if it is None. Then is there any condition when `has_local_source_paths_file_mounts` would be False? Perhaps it should be a bool check on `copy_mounts`.